### PR TITLE
[WX-1700] Fix to string & optional concatenation

### PR DESCRIPTION
**Pull request by THWiseman**
### Description

Cromwell will now correctly emit nothing when concatenating a string with an empty optional, as outlined in the [WDL spec](https://github.com/openwdl/wdl/blob/main/versions/1.0/SPEC.md#prepending-a-string-to-an-optional-parameter).

Before:
`"Something something ${"hello" + myEmptyOption}` -> throws either `"Sorry! Operation + is not supported on empty optional values."` or `"No implementation of FileEvaluator[StringExpression]"`. 

After: 
`"Something something ${"hello" + myEmptyOption}` correctly evaluates to the string `"Something something"`. 

#### `CHANGELOG.md`
 - [x] I updated `CHANGELOG.md` in this PR
 - [ ] I assert that this change shouldn't be included in `CHANGELOG.md` because it doesn't impact community users

#### Terra Release Notes
 - [x] I added a suggested release notes entry in this Jira ticket
 - [ ] I assert that this change doesn't need Jira release notes because it doesn't impact Terra users

### CODE REVIEW

File: wdl/transforms/new-base/src/main/scala/wdl/transforms/base/linking/expression/files/LiteralEvaluators.scala [hunk 1]
```diff
@@ -107,4 +109,24 @@ object LiteralEvaluators {
...
+      val predictedFiles: Seq[ErrorOr[Set[WomFile]]] = a.pieces.map {
+        case literal: StringLiteral =>
+          stringLiteralEvaluator.predictFilesNeededToEvaluate(literal, inputs, ioFunctionSet, coerceTo)
+        case _ => Valid(Set.empty[WomFile])
+      }
+      val sets: Set[WomFile] = predictedFiles.collect { case Valid(set) => set }.foldLeft(Set[WomFile]())(_ ++ _)
+      val errors: Seq[NonEmptyList[String]] = predictedFiles.collect { case Invalid(error) => error }
+      errors.headOption.map(head => head.invalid).getOrElse(sets.validNel)
+    }
+  }
```
Comment (jgainerdewar) [inline wdl/transforms/new-base/src/main/scala/wdl/transforms/base/linking/expression/files/LiteralEvaluators.scala:131]: Can we add some unit testing of this? Looks like we already have a `FileEvaluatorSpec`.

File: wdl/transforms/draft3/src/test/scala/wdl/draft3/transforms/linking/expression/files/FileEvaluatorSpec.scala [hunk 2]
```diff
@@ -26,4 +26,20 @@ class FileEvaluatorSpec extends AnyFlatSpec with CromwellTimeoutSpec with Matche
...
+
+    val expressionElement: ExpressionElement =
+      StringExpression(Seq(filePiece, anotherFilePiece, notAFilePiece, anotherNotAFilePiece))
+
+    val evaluatedFiles = expressionElement.predictFilesNeededToEvaluate(inputs = Map.empty,
+                                                                        ioFunctionSet = NoIoFunctionSet,
+                                                                        coerceTo = WomSingleFileType
+    )
+    evaluatedFiles shouldBeValid Set(WomSingleFile("moo.txt"), WomSingleFile("boo.txt"))
+  }
```
Comment (jgainerdewar) [inline wdl/transforms/draft3/src/test/scala/wdl/draft3/transforms/linking/expression/files/FileEvaluatorSpec.scala:44]: Do you think it makes sense to also test the error case here, or whatever it looks like in this function when the optional variable is null?

**Review by jgainerdewar — APPROVED** [review]

**Review by aednichols — APPROVED** [review]
Nice work! We should definitely let David know.
